### PR TITLE
Add `ir_utils::createRangeLoop`

### PR DESCRIPTION
**Pull request by zasdfgbnm**
This utility will be used to implement warp specialization.

### CODE REVIEW

File: csrc/device_lower/pass/allocation.cpp [hunk 1]
```diff
@@ -670,8 +652,12 @@ class AllocationInserter : public kir::ExprMutator {
       kir::Allocate* mbarrier_alloc =
           IrBuilder::create<kir::Allocate>(mbarrier, MemoryType::Shared);
 
-      auto mbarrier_init = initializeMbarrier(fl, mbarrier);
-      auto mbarrier_inval = invalidateMbarrier(fl, mbarrier);
+      // Initialize and invalidate of mbarriers that are used to notify the
+      // completion of loading of the circular buffer.
+      auto mbarrier_init_filled =
```
Comment (zasdfgbnm) [inline csrc/device_lower/pass/allocation.cpp:657]: Just an unrelated variable renaming.

**Comment by zasdfgbnm** [comment]
!test

**Comment by zasdfgbnm** [comment]
!test

**Review by rdspring1 — APPROVED** [review]
